mesa/space.py

```diff
@@ -485,7 +486,463 @@ def exists_empty_cells(self) -> bool:
+
+        if not eligible_cells:
+            warn(
+                f"No eligible cells found. Agent {agent.unique_id} remains in the current position.",
+                RuntimeWarning,
+                stacklevel=2,
+            )
+            return  # Agent stays in the current position
+
+        new_pos = agent.random.choice(eligible_cells)
+        self.move_agent(agent, new_pos)
+
+    def select_extreme_value_cells(
+        self,
+        property_name: str,
+        mode: str,
+        mask: np.ndarray = None,
+        return_list: bool = True,
+    ) -> list[Coordinate] | np.ndarray:
+        """
+        Select cells with the highest or lowest property value, optionally with a mask.
+
+        Args:
+            property_name (str): The name of the property layer.
+            mode (str): 'highest' or 'lowest'.
+            mask (np.ndarray, optional): A boolean mask to restrict the selection.
+            return_list (bool, optional): If True, return a list of coordinates, otherwise return a mask.
+
+        Returns:
+            Union[list[Coordinate], np.ndarray]: List of coordinates or a boolean mask of cells with the extreme property value.
+        """
+        prop_values = self.properties[property_name].data
+        if mask is not None:
+            prop_values = np.where(mask, prop_values, np.nan)
+
+        if mode == "highest":
+            extreme_value = np.nanmax(prop_values)
+        elif mode == "lowest":
+            extreme_value = np.nanmin(prop_values)
+        else:
+            raise ValueError(f"Invalid mode {mode}. Choose from 'highest' or 'lowest'.")
+
+        # Optimize the mask creation using numpy's inherent functions
+        target_mask = prop_values == extreme_value
+
+        if return_list:
+            return list(zip(*np.where(target_mask)))
+        else:
+            return target_mask
+
+    def move_agent_to_extreme_value_cell(
+        self, agent: Agent, property_name: str, mode: str, mask: np.ndarray = None
+    ) -> None:
+        """
+        Move an agent to a cell with the highest or lowest property value,
+        optionally with a mask.
+
+        Args:
+            agent (Agent): The agent to move.
+            property_name (str): The name of the property layer.
+            mode (str): 'highest' or 'lowest'.
+            mask (np.ndarray, optional): A boolean mask to restrict the selection.
+        """
+        target_cells = self.select_extreme_value_cells(
+            property_name, mode, mask, return_list=True
+        )
+
+        # If no eligible cells are found, issue a warning and keep the agent in its current position.
+        if len(target_cells) == 0:
+            warn(
+                f"No eligible cells found. Agent {agent.unique_id} remains in the current position.",
+                RuntimeWarning,
+                stacklevel=2,
+            )
+            return  # Agent stays in the current position
+
+        new_pos = tuple(agent.random.choice(target_cells))
```

Comment:
This might be surprising that a cell is chosen randomly

Reply:
What would your alternative be, if we have multiple cells with the same highest or lowest value?